packages/vm/src/errors.rs

```diff
@@ -99,28 +102,120 @@ impl From<wasmer_runtime_core::cache::Error> for VmError {
     }
 }
 
+impl From<wasmer_runtime_core::error::CompileError> for VmError {
+    fn from(original: wasmer_runtime_core::error::CompileError) -> Self {
+        make_compile_err(format!("Wasmer compile error: {:?}", original))
+    }
+}
+
+impl From<wasmer_runtime_core::error::ResolveError> for VmError {
+    fn from(original: wasmer_runtime_core::error::ResolveError) -> Self {
+        make_resolve_err(format!("Wasmer resolve error: {:?}", original))
+    }
+}
+
+impl From<wasmer_runtime_core::error::RuntimeError> for VmError {
+    fn from(original: wasmer_runtime_core::error::RuntimeError) -> Self {
+        make_runtime_err(format!("Wasmer runtime error: {:?}", original))
+    }
+}
+
 pub type VmResult<T> = core::result::Result<T, VmError>;
 
+pub fn make_backend_err<S: Into<String>>(msg: S) -> VmError {
+    BackendErr { msg: msg.into() }.build()
+}
+
 pub fn make_cache_err<S: Into<String>>(msg: S) -> VmError {
     CacheErr { msg: msg.into() }.build()
 }
 
+pub fn make_compile_err<S: Into<String>>(msg: S) -> VmError {
+    CompileErr { msg: msg.into() }.build()
+}
+
+pub fn make_conversion_err<S: Into<String>, T: Into<String>, U: Into<String>>(
```

Comment:
In the README, you mentioned these are not exposed outside the crate, people that import this can only use them.

How about making them all `pub(crate)` to make that explicit in the code.

Reply:
What I wrote in the README is probably not 100 % accurate since even before the error creator structs were not exported because of `pub use crate::errors::{VmError, VmResult};` in `lib.rs`.

I think it is sufficient to not export them in `lib.rs`. This should really be the standard case: only export explicitely with good reason. Then you only need `pub(crate)` to hide some functions in otherwise exported structs.